sxt/multiexp/pippenger2/variable_length_computation.cc

```diff
@@ -0,0 +1,60 @@
+/** Proofs GPU - Space and Time's cryptographic proof algorithms on the CPU and GPU.
+ *
+ * Copyright 2024-present Space and Time Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sxt/multiexp/pippenger2/variable_length_computation.h"
+
+#include <algorithm>
+
+#include "sxt/base/error/assert.h"
+
+namespace sxt::mtxpp2 {
+//--------------------------------------------------------------------------------------------------
+// compute_product_length_table
+//--------------------------------------------------------------------------------------------------
+void compute_product_length_table(basct::span<unsigned>& product_lengths,
+                                  basct::cspan<unsigned> bit_widths,
+                                  basct::cspan<unsigned> output_lengths, unsigned first,
+                                  unsigned length) noexcept {
+  auto num_products = product_lengths.size();
+  auto num_outputs = bit_widths.size();
+  SXT_DEBUG_ASSERT(
+      // clang-format off
+      num_products >= num_outputs &&
+      product_lengths.size() == num_products &&
+      bit_widths.size() == num_outputs &&
+      output_lengths.size() == num_outputs
+      // clang-format on
+  );
+
+  // find first output with longer than <length>
```

Comment:
NIT: I'm not sure I understand this comment? Wouldn't `std::count_if` return a count of all items in the `output_lengths` span that are `<= first`? Should `first` be `length` in the predicate?

Reply:
I changed the comment. Thank you.